admin/cluster_setup.py

```diff
@@ -177,66 +177,64 @@ def main(reactor, args, base_path, top_level):
     reactor.addSystemEventTrigger(
         'before', 'shutdown', log_writer.stopService)
 
-    cluster = None
-    results = []
-    try:
-        yield runner.ensure_keys(reactor)
-        cluster = yield runner.start_cluster(reactor)
+    def cluster_cleanup():
+        print("stopping cluster")
+        return runner.stop_cluster(reactor)
 
-        managed_config_file = options['cert-directory'].child("managed.yaml")
-        managed_config = create_managed_config(options['config'], cluster)
-        managed_config_file.setContent(
-            yaml.safe_dump(managed_config, default_flow_style=False)
-        )
+    cleanup_trigger_id = reactor.addSystemEventTrigger('before', 'shutdown',
+                                                       cluster_cleanup)
```

Comment:
Triggers run in the order they're added.  I bet it would be good if we did the cluster cleanup before we stopped the log writer rather than after.